src/client/Components/Util/OAuth.js

```diff
@@ -0,0 +1,92 @@
+import React from 'react';
+import Button from '@material-ui/core/Button';
+import PropTypes from 'prop-types';
+
+export default class OAuth extends React.Component {
+  state = {
+    user: {},
+    disabled: ''
+  };
+
+  constructor(props) {
+    super(props);
+
+    this.startAuth = this.startAuth.bind(this);
+    this.closeCard = this.closeCard.bind(this);
+  }
+
+  componentDidMount() {
+    const { socket, provider } = this.props;
+
+    socket.on(provider, (user) => {
+      this.popup.close();
+      this.setState({ user });
+    });
+  }
+
+  checkPopup() {
+    const check = setInterval(() => {
+      const { popup } = this;
+      if (!popup || popup.closed || popup.closed === undefined) {
+        clearInterval(check);
+        this.setState({ disabled: '' });
+      }
+    }, 1000);
+  }
+
+  openPopup() {
+    const { provider, socket } = this.props;
```

Comment:
I'm guessing this is destructuring syntax, which you seem to be in love with. Can we use `this.$VAR` when we only use the value once? (Unless there is some benefit I'm missing)

Reply:
Our ESLint settings yell at me if I don't destructure...I can use this.$VAR I  just have to be ok with the red squiggle lines that come with it